src/dashboard-modal.tsx

```diff
@@ -0,0 +1,191 @@
+              focused: styles["dashboardUrl"],
+              notchedOutline: styles["dashboardUrl"]
+            },
+            endAdornment: (
+              <InputAdornment position="end">
+                <ClickAwayListener onClickAway={this.closeTooltip}>
+                  <div>
+                    <Tooltip title="Copied!" placement="top" onClose={this.closeTooltip} open={this.state.showTooltip} PopperProps={{ disablePortal: true }} disableFocusListener={true} disableHoverListener={true} disableTouchListener={true}>
+                      <CopyToClipboard text={this.state.dashboardUrl} onCopy={this.copyDashboardUrl}>
+                        <IconButton color="primary" aria-label="Copy dashboard URL" onClick={this.copyDashboardUrl}>
+                          <svg aria-hidden="true" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 896 1024" height="24"><path d="M128 768h256v64H128v-64z m320-384H128v64h320v-64z m128 192V448L384 640l192 192V704h320V576H576z m-288-64H128v64h160v-64zM128 704h160v-64H128v64z m576 64h64v128c-1 18-7 33-19 45s-27 18-45 19H64c-35 0-64-29-64-64V192c0-35 29-64 64-64h192C256 57 313 0 384 0s128 57 128 128h192c35 0 64 29 64 64v320h-64V320H64v576h640V768zM128 256h512c0-35-29-64-64-64h-64c-35 0-64-29-64-64s-29-64-64-64-64 29-64 64-29 64-64 64h-64c-35 0-64 29-64 64z"/></svg>
+                        </IconButton>
+                      </CopyToClipboard>
+                    </Tooltip>
+                  </div>
+                </ClickAwayListener>
+              </InputAdornment>
+            )
+          }}
+          InputLabelProps={{
+            FormLabelClasses: {
+              root: styles["dashboardUrl"],
+              focused: styles["dashboardUrl"]
+            }
+          }}
+        />
+      );
+    }
+    return null;
+  }
+
+  private handleChange(selectedOptions: any) {
+    this.setState({
+      selectedOptions,
+      hasCreatedDashboard: false
+    });
+  }
+
+  private getNames() {
+    this.setState({ loading: true });
+    API.getNames((names) => {
+      nameOptions = this.parseNameData(names);
+      this.setState({ loading: false });
+    });
+  }
+
+  private parseNameData(namesData: NameData[]){
+    return namesData.map(nameData => {
+      const nameOption: NameOption = {
+        value: nameData.name,
+        label: nameData.name
+      };
+      return nameOption;
+    });
+  }
+
+  private createDashboardUrl() {
+    let url = encodeURI(window.location.origin + "?q=");
+    let hasBrokenLimit = false;
+    let firstIgnoredItem = "";
+
+    for (const option of this.state.selectedOptions) {
+      const encodedOption = encodeURI(option.value);
+      if (url.length <= (maxUrlSize - encodedOption.length)) {
+        url += encodedOption + encodeURI("|");
+      } else {
+        hasBrokenLimit = true;
+        firstIgnoredItem = option.value;
+        break;
+      }
+    }
+
+    this.setState({
+      dashboardUrl: url.substring(0, url.length - 3),
```

Comment:
Add a comment to explain the substring and the ` - 3` since it's not immediately clear